models/contacts.js

```diff
@@ -1,19 +1,88 @@
-// const fs = require('fs/promises')
+const fs = require('fs/promises');
+const path = require('path');
+const Joi = require('joi');
 
-const listContacts = async () => {}
+const contactsFilePath = path.join(__dirname, 'contacts.json');
 
-const getContactById = async (contactId) => {}
+const listContacts = async () => {
+  const data = await fs.readFile(contactsFilePath, 'utf-8');
+  return JSON.parse(data);
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
+  const contacts = await listContacts();
+  const contact = contacts.find((contact) => contact.id === contactId);
 
-const addContact = async (body) => {}
+  if (!contact) {
+    throw new Error('Contact not found');
+  } else {
+    return contact;
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+const removeContact = async (contactId) => {
+  const contacts = await listContacts();
+  const updatedContacts = contacts.filter(
+    (contact) => contact.id !== contactId
+  );
+
+  await fs.writeFile(contactsFilePath, JSON.stringify(updatedContacts));
+};
+
+const validateContact = (contact) => {
+  const schema = Joi.object({
+    name: Joi.string().required(),
+    email: Joi.string().email().required(),
+    phone: Joi.string().required(),
+  });
+
+  return schema.validate(contact);
+};
+
+const addContact = async (name, email, phone) => {
+  const contacts = await listContacts();
+  const newContact = {
+    id: String(Date.now()),
```

Comment:
better use libraty like `uuid` for creating unique `ids`

Reply:
I'm not gonna lie i was checking others repos and i saw uuid library but 'cause of honeymoon i hadn't got time to read the files  i'll check it when i'll be back in Poland. Thank You